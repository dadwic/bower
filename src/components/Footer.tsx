import React from "react";
import Box from "@mui/material/Box";
import Link from "@mui/material/Link";
import Fab from "@mui/material/Fab";
import Typography from "@mui/material/Typography";
import Container from "@mui/material/Container";
import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
import ScrollTop from "./ScrollTop";

export interface Props {
  window?: () => Window;
  children?: React.ReactElement | null;
}

function Copyright() {
  return (
    <Typography variant="body2" color="text.secondary">
      {`Copyright © ${new Date().getFullYear()}`}
    </Typography>
  );
}

export default function Footer(props: Props) {
  return (
    <Box
      component="footer"
      sx={{
        py: 3,
        px: 2,
        mt: "auto",
        backgroundColor: (theme) =>
          theme.palette.mode === "light"
            ? theme.palette.grey[200]
            : theme.palette.grey[800],
      }}
    >
      <Container maxWidth="sm">
        <Typography variant="body1">
          Made with ❤️ by{" "}
          <Link color="inherit" href="http://dadwic.com/" target="_blank">
            @dadwic
          </Link>
        </Typography>
        <Copyright />
      </Container>
      <ScrollTop {...props}>
        <Fab color="info" size="small" aria-label="scroll back to top">
          <KeyboardArrowUpIcon />
        </Fab>
      </ScrollTop>
    </Box>
  );
}
