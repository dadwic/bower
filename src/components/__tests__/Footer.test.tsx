import React from "react";
import renderer from "react-test-renderer";
import Footer from "../Footer";

describe("Footer", () => {
  it("should corretly match snapshot", () => {
    const tree = renderer.create(<Footer />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
