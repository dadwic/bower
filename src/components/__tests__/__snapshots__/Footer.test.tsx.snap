// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should corretly match snapshot 1`] = `
<footer
  className="MuiBox-root css-9jyc04"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Made with ❤️ by
       
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1kgfegd-MuiTypography-root-MuiLink-root"
        href="http://dadwic.com/"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        @dadwic
      </a>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      Copyright © 2022
    </p>
  </div>
  <div
    className="MuiBox-root css-mgur8"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "transform": "scale(0)",
        "visibility": "hidden",
      }
    }
  >
    <button
      aria-label="scroll back to top"
      className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-info css-10ma2jt-MuiButtonBase-root-MuiFab-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowUpIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
        />
      </svg>
    </button>
  </div>
</footer>
`;
